server/start.go

```diff
@@ -310,30 +313,46 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 	}
 
 	genDocProvider := node.DefaultGenesisDocProviderFunc(cfg)
-	tmNode, err := node.NewNode(
-		cfg,
-		pvm.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile()),
-		nodeKey,
-		proxy.NewLocalClientCreator(app),
-		genDocProvider,
-		node.DefaultDBProvider,
-		node.DefaultMetricsProvider(cfg.Instrumentation),
-		ctx.Logger.With("server", "node"),
+	var (
+		tmNode *node.Node
+		modes  = strings.Split(ctx.Viper.GetString(srvflags.QueryOnly), ",")
```

Comment:
If converted flag to `bool`, wouldn't need to do this split (wouldn't this cause issues if not inputted correctly?)